### DISCUSSION TITLE
How to overlay many .pngs over many .jpgs in a folder?

### DISCUSSION THREAD

**Discussion by camilapaleno** [ImageMagick/ImageMagick#7470] [Help]
### ImageMagick version

7.1.1-34 

### Operating system, version and so on

macOS Ventura

### Description

The file names look like this: 

 - 363-main.jpg 
 - 363-overlay.png  
 - 364-main.jpg 
 - 365-main.jpg
 - 365-overlay.png
 - 366-main.jpg
 - 366-overlay.png
 - 367-main.jpg
 - 368-main.jpg

So there are many .jpgs that do not have a matching .png overlay that will need to be skipped. All overlays are unique to the main image.

This is what I tried (pulled from here: https://www.imagemagick.org/discourse-server/viewtopic.php?t=32333)

    magick *-main.jpg -extent %[fx:s.w*2] null: *-overlay.png -gravity east -layers composite _%03d_out.png

And this is the error I get:

    no matches found: %[fx:s.w*2]

Can anyone provide any clarity on why I'm getting this error and how I can fix this? Thank you.

**Comment by fmw42** [comment]
See answers on your StackOverflow same question.
